fake_spectra/voigtfit.py

```diff
@@ -57,36 +57,38 @@ def do_fit(self, tol=1e-4, signif = 0.9, sattau = 4):
         fitted_tau = np.array(self.tau)
         #We do not care about very small peaks; this includes both peaks which are small in absolute value
         #and peaks which are small in a relative sense.
-        realtol = tol * np.max([1,np.max(self.tau)])
+        realtol = tol * np.max([1, np.max(self.tau)])
         #Initial mask includes the whole spectrum
         mask = np.where(fitted_tau > -1)
         #Do the fit iteratively, stopping when the largest peak is likely unimportant.
-        while np.max(fitted_tau[mask]) > realtol:
-            (fitted_tau, mean, amplitude, stddev)=self.iterate_new_spectrum(fitted_tau, mask=mask)
+        while np.max(fitted_tau[mask]) > realtol and mask[0].size != 0:
```

Comment:
The initial mask is just to get the while loop started.

The check on if the there are actually any points in the mask is there for later iterations once the mask has been adjusted to include only sections that are not over-saturated with Voigt profiles (so not necessarily related to the -1 value). As the profiles get subtracted it can run into this, but only at higher redshift where it is fitting lots of tall peaks

Reply:
I see, thanks!